Coding-Coach/coding-coach
# Migrate Panel component to tailwind #238

# Details
Updated the Panel component to use the Tailwind CSS.
* Created plugin for the panel-content, containing selectors to select the elements rendered inside of the content part of the panel.
* Updated snapshots and tests
* Removed unneeded files, i.e. relevant .scss files.
 
## Description
This PR updates the already existing panel component to use Tailwind CSS.  Includes the changes mentioned above.

### Updates since last review
* Deleted unneeded changes to tailwind.js.
* Added a 5rem to minHeight and 9rem to height.  Both of these are needed to maintain the same design for panel that already existed.  ... before header was 75px, footer was 65px, and content was 100% - (75px + 65px), now we need 100% - (5rem + 4rem) which is where the 9rem comes from.  
* If wanted, I can remove the 5rem from minHeight and add it as an inline style, but I don't think we can escape adding the 9 rem because if the header and footer heights don't add up the alignment/spacing gets thrown off.

## Steps to QA
1.  Load my PR code and load https://codingcoach.io/ side by side.
2. Scroll to the bottom and click all the buttons Terms & Conditions, Cookies, Privacy Policy, and compare localhost to the active site and validate they look the same.
3. Test the different screen sizes and different browsers to make sure it looks correct on all.

## Screenshots
<img width="841" alt="screen shot 2018-11-30 at 7 38 04 pm" src="https://user-images.githubusercontent.com/22843948/49322907-c382a600-f4d9-11e8-96d6-016ed0726be2.png">
<img width="841" alt="screen shot 2018-11-30 at 7 37 50 pm" src="https://user-images.githubusercontent.com/22843948/49322908-c382a600-f4d9-11e8-95f1-8df745211634.png">
<img width="841" alt="screen shot 2018-11-30 at 7 37 46 pm" src="https://user-images.githubusercontent.com/22843948/49322909-c382a600-f4d9-11e8-8c2e-2a1e8ad03faa.png">
<img width="841" alt="screen shot 2018-11-30 at 7 37 27 pm" src="https://user-images.githubusercontent.com/22843948/49322910-c382a600-f4d9-11e8-9010-8882edc8ea8d.png">

## Issue
Closes #238
